src/Servers/IIS/IIS/src/Core/IISHttpContext.FeatureCollection.cs

```diff
@@ -499,5 +502,15 @@ private void DisableCompression()
             var serverVariableFeature = (IServerVariablesFeature)this;
             serverVariableFeature["IIS_EnableDynamicCompression"] = "0";
         }
+
+        void IConnectionLifetimeNotificationFeature.RequestClose()
+        {
+            // Set the connection close feature if the response hasn't sent headers as yet
+            if (!HasResponseStarted)
+            {
+                ResponseHeaders.Connection = ConnectionClose;
+            }
```

Comment:
Since IIS never supported this feature via headers, let's skip setting the header and instead try calling `NativeMethods.HttpSetNeedGoAway(_requestNativeHandle);` directly. You'll still need the HasResponseStarted check since GoAway is implemented in Http.Sys as a response flag.

Reply:
Wait, we still need it for HTTP/1.1 don't we. Nevermind, this is fine.